### PR TITLE
lite renames and refactors using ES6

### PR DESCRIPTION
**Pull request by staltz**
A bunch of light-weight refactors such as:

- Use arrow functions that are more concise
- Put names on anonymous functions so that they are more debuggable in stack traces
- Use `{ name, value }` instead of `{ name: name, value: value }`
- Use `if (cb) cb()` instead of `cb && cb()`
- Use string interpolation

### CODE REVIEW

File: pull-weird.js [hunk 1]
```diff
@@ -46,10 +43,11 @@ module.exports = function (weird, _done) {
   }
 
   return {
-    source: function (abort, cb) {
+    source (abort, cb) {
```
Comment (staltz) [inline pull-weird.js:46]: Not only is this pattern shorter, it makes sure that the function is not anonymous. `this` support is preserved

File: remote-api.js [hunk 2]
```diff
@@ -2,26 +2,15 @@
...
   for (const name in manifest) {
-    (function (name, type) {
-      const _path = path ? path.concat(name) : [name]
-      obj[name] =
-        isObject(type)
-          ? recurse({}, type, _path, remoteCall)
-          : function () {
-            return remoteCall(type, _path, [].slice.call(arguments))
-          }
-    })(name, manifest[name])
```
Comment (staltz) [inline remote-api.js:24]: I took a careful look at the git history, and I don't understand why was this block wrapped in an IIFE. I thought about it, and it seems to make sense to not have the IIFE inside the for loop

File: stream.js [hunk 3]
```diff
@@ -47,33 +35,37 @@ module.exports = function initStream (localCall, codec, onClose) {
...
-          value = localCall(type, name, data.args)
+          value = localCall(type, name, args)
         } catch (_err) {
           err = _err
         }
 
-        const _stream = pullWeird[
-          { source: 'sink', sink: 'source' }[type] || 'duplex'
-        ](stream)
+        const antiType =
```
Comment (arj03) [inline stream.js:58]: great naming here

File: pull-weird.js (hunk 1 above)
Comment (arj03) [reply pull-weird.js:46]: Yeah I was looking at that and thinking, ahh that is actually a great usage of the `{ a = a` pattern. Good to know about this.

**Review by arj03 — APPROVED** [review]

**Pull request merged**